### ISSUE TITLE
Provide icx value input in contract explorer

### ISSUE THREAD

**Issue by nightowl121** [sudoblockio/icon-tracker-frontend#435]
If the method used in the contract explorer is payable method, the contract explorer needs to provide a field to input the ICX value you want to send in the method.

Currently the explorer doesn't provide this option.

**Comment by robcxyz** [comment]
Hi @nightowl121 - I like this idea but to be clear, the contract explorer is based off pulling the contract's ABI and from that perspective, I don't think there is anything to distinguish an input value as being payable vs a non-payable. I could be wrong, this is not my domain as much. Someone else would know better. 

But if that is the case, this is not possible to make as a default within the contract explorer but rather could be some toggle-able item that denominates in ICX (ie *1e18). This could also be left to another section that supports token transfers and other types of payment utilities. 

If I am wrong though about payable not being in the ABI please let me know otherwise any feedback on the above is appreciated.

**Comment by nightowl121** [comment]
Here is an example of a method in ABI showing that the function is payable.
https://tracker.icon.community/contract/cx21e94c08c03daee80c25d8ee3ea22a20786ec231#code
`{
		"inputs": [
			{
				"name": "_path",
				"type": "[]Address"
			},
			{
				"default": "0x0",
				"name": "_minReceive",
				"type": "int"
			},
			{
				"default": null,
				"name": "_receiver",
				"type": "str"
			}
		],
		"name": "route",
		"outputs": [],
		"payable": "0x1",
		"type": "function"
	}`

**Comment by robcxyz** [comment]
Ah got it. But then if there are multiple int fields, is it possible to differentiate between an icx denominated value vs standard int? Also does payable strictly define icx transfers or could they be token transfers as well which could have different decimals and can't be assumed?

Don't get me wrong, we should do something and not trying to figure out a reason to not incorporate this somewhere, just can't make assumptions which here and if payables / decimals etc need to be accounted for, we can build a separate tool for that

**Comment by nightowl121** [comment]
Payable is in it's own a different field or parameter. So if there is a payable function, you should add a different input box mentioning the amount of icx to send in the transaction. You can either take the input in ICX or loop (10^-18 ICX) whichever will give better UX. Etherscan allows option to add 18 zeros at once. Other parameters remain same as it is.

Payable will always be for ICX. It won't handle any other token transfers at least in ICON.

In short, if this feature has to be integrated in the current contract explorer, the additional logic would be as:
- Check if a method is payable. For payable method, add an additional input box mentioning the amount of icx to send. 0 if not provided by user.
- For rest methods, continue as it is.

**Comment by robcxyz** [comment]
@nightowl121 - I like this idea. So basically any `payable` method should by default be in ICX and then perhaps have a toggle between `Loop` and `ICX` here. 

![image](https://github.com/sudoblockio/icon-tracker-frontend/assets/6512972/f79f6b3a-cc65-4133-a564-24581d3f13d7)

Would be like a radio button next to `Query`

Agree?

**Comment by nightowl121** [comment]
Yeah sounds good.

**Comment by nightowl121** [comment]
Is this done only for a specific contract? I can't see this happening on other contract methods. 
@athoifss 

<img width="408" alt="Image" src="https://github.com/user-attachments/assets/4c8fbe76-85d7-498e-8560-9d5af4060a9e" />

<img width="290" alt="Image" src="https://github.com/user-attachments/assets/b38d0fcf-7520-4c63-969b-0af278c0df70" />

<img width="281" alt="Image" src="https://github.com/user-attachments/assets/10bb2ddb-9635-4c6f-b638-2157067440f4" />

**Comment by athoifss** [comment]
@nightowl121 sorry did not see your comment earlier. 


I can see the value field for other contracts as well. For example,  from here https://tracker.icon.community/contract/cx43e2eec79eb76293c298f2b17aec06097be606e0#contract
![Image](https://github.com/user-attachments/assets/f35cc718-dfed-49a7-8a2c-f2a31efbec41)

Can you please tell me which contract is the above screnshot from (registerProposal) ?
Maybe it already has a value field (of type 'bytes') ?

**Comment by nightowl121** [comment]
It is from this contract. It's registerProposal method. 
[cx0000000000000000000000000000000000000001](https://tracker.icon.community/contract/cx0000000000000000000000000000000000000001#contract)

You are right that the method already contains the `value` parameter.

**Comment by athoifss** [comment]
@FidelVe @nightowl121 @robcxyz 
So what do can be done about this ? A method which already has a value param but is also payable ?

**Comment by FidelVe** [comment]
there is no issue because each key `value` goes in a different place in the JSON-RPC, this example can illustrate it better:

```json
{
  "to": "cx0000000000000000000000000000000000000001",
  "from": "hx0169e0...f063",
  "stepLimit": "0x9502f900",
  "nid": 2,
  "version": "0x3",
  "timestamp": 1740063937487000,
  "value": "0x56bc75e2d63100000", // this value refers to the amount of ICX to send in this transaction
  "nonce": "0x1",
  "dataType": "call",
  "data": {
    "method": "registerProposal",
    "params": {
      "title": "proposal title",
      "description": "proposal description",
      "value": "0x5b7b226e6...7d7d5d" // this is just a key in the object named 'value' and is to sending the bytecode of the proposal
    }
  }
}
╭
```
